File: server/SocialMedia/SocialMedia.Domain/Entities/Post.cs

namespace SocialMedia.Domain.Entities
{
    public class Post
    {
        public int Id { get;  set; } 
        public string Content { get;  set; }      
        public int AuthorUserId { get;  set; }
        public DateTime CreatedAt { get; set; }
        public int RepostCount { get;  set; } = 0;
        public bool IsRepost { get;  set; } = false;
        public int? OriginalPostId { get;  set; }

        public virtual Post? OriginalPost { get; set; }
        public virtual User User { get; set; }

        public Post()
        {
        }

        public void IncrementRepost()
        {
            RepostCount++;
        }
    }
}

File: web/social-media/src/type/api/post.ts
export interface OriginalPostDTO {
    id: number;
    content: string;
    authorNickname: string;
    authorProfileImageUrl: string;
    authorUserId: number;
    createdAt: string;
    repostCount: number;
    isRepost: boolean;
  }
  
  export interface PostResponseDTO {
    id: number;
    content: string;
    authorNickname: string;
    authorProfileImageUrl: string;
    authorUserId: number;
    createdAt: string;
    repostCount: number;
    isRepost: boolean;
    originalPost?: OriginalPostDTO | null;
  }
  
  export interface PaginatedPosts {
    items: PostResponseDTO[];
    totalCount: number;
    page: number;
    pageSize: number;
  }

  export enum PostOrderBy {
    Latest = 1,
    Trending = 2,
  }